src/common/file_utils.cpp

```diff
@@ -4,15 +4,64 @@
 #include "common/string_utils.h"
 #include "glob/glob.hpp"
 
+#ifdef _WIN32
+#include <fileapi.h>
+#include <windows.h>
+#endif
+
 namespace kuzu {
 namespace common {
 
+FileInfo::~FileInfo() {
+#ifdef _WIN32
+    if (handle != nullptr) {
+        CloseHandle((HANDLE)handle);
+    }
+#else
+    if (fd != -1) {
+        close(fd);
+    }
+#endif
+}
+
+int64_t FileInfo::getFileSize() {
+#ifdef _WIN32
+    return GetFileSize((HANDLE)handle, nullptr);
+#else
+    struct stat s;
+    if (fstat(fd, &s) == -1) {
+        return -1;
+    }
+    return s.st_size;
+#endif
+}
+
 std::unique_ptr<FileInfo> FileUtils::openFile(const std::string& path, int flags) {
+#if defined(_WIN32)
+    // Not providing GENERIC_READ seems to cause problems.
+    int dwDesiredAccess = GENERIC_READ;
+    int dwCreationDisposition = OPEN_EXISTING;
+    int dwShareMode = FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE;
+    int dwFlagsAndAttributes = FILE_ATTRIBUTE_NORMAL;
+    if (flags & O_CREAT)
+        dwCreationDisposition = OPEN_ALWAYS;
```

Comment:
I think you can rewrite this as `auto dwCreationDisposition = (flags & O_CREAT) ?  OPEN_ALWAYS : OPEN_EXISTING`